### PR TITLE
tassel: add version 3.0

### PR DESCRIPTION
**Pull request by downloadico**

### CODE REVIEW

File: var/spack/repos/builtin/packages/tassel/package.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class Tassel(Package):
     homepage = "https://www.maizegenetics.net/tassel"
     git      = "https://bitbucket.org/tasseladmin/tassel-5-standalone.git"
 
+    version('3.0', git='git://git.code.sf.net/p/tassel/tassel3-standalone.git')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/tassel/package.py]: Can we download a tarball, or fix to a commit sha if we are dealing with a fixed version?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/tassel/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/tassel/package.py]: fixed: I gave it a more specific version number too.

File: var/spack/repos/builtin/packages/tassel/package.py [hunk 2]
```diff
@@ -13,6 +13,7 @@ class Tassel(Package):
     homepage = "https://www.maizegenetics.net/tassel"
     git      = "https://bitbucket.org/tasseladmin/tassel-5-standalone.git"
 
+    version('3.0.174', commit='612a92c0f677d6ec040c17e146671974a3cdc2da', git='git://git.code.sf.net/p/tassel/tassel3-standalone.git')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/tassel/package.py]: Reading [here](https://bytebucket.org/tasseladmin/tassel-5-source/wiki/docs/UpdatingTasselStandaloneUsingGit.pdf) I think this package might need some reworking. I concur the version number you used is better, but then my understanding is that the `2017-07-22` version is in the 5.X series, right? Can we use consistent version numbers?

Comment (alalazo) [reply var/spack/repos/builtin/packages/tassel/package.py]: Also, it seems to me that the constraints on `java` are not encoded in the `package.py`.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/tassel/package.py (hunk 2 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/tassel/package.py]: the 2017-07-22 version was in the package.py before I touched it.  I just updated it in my latest PR update after tracking down that commit... 

the tassel package depends on java, what constraints are you referring to?

Comment (alalazo) [reply var/spack/repos/builtin/packages/tassel/package.py]: > the tassel package depends on java, what constraints are you referring to?

In the link above there are mentions of the required Java versions which are not encoded in `package.py`

File: var/spack/repos/builtin/packages/tassel/package.py [hunk 3]
```diff
@@ -13,7 +13,8 @@ class Tassel(Package):
     homepage = "https://www.maizegenetics.net/tassel"
     git      = "https://bitbucket.org/tasseladmin/tassel-5-standalone.git"
 
-    version('2017-07-22', commit='ae96ae75c3c9a9e8026140b6c775fa4685bdf531')
+    version('3.0.174', commit='612a92c0f677d6ec040c17e146671974a3cdc2da', git='git://git.code.sf.net/p/tassel/tassel3-standalone.git')
+    version('5.2.39', commit='ae96ae75c3c9a9e8026140b6c775fa4685bdf531')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/tassel/package.py]: Can you move this version declaration at the top (so that versions are in descending order)?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a couple of minor comments, otherwise LGTM

File: var/spack/repos/builtin/packages/tassel/package.py (hunk 3 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/tassel/package.py]: Done.

**Comment by snehring** [comment]
Is this dead? I just received a request to update this from one of our users.

**Pull request closed without merging**